nav2_dwb_controller/dwb_critics/test/obstacle_footprint_test.cpp

```diff
@@ -125,11 +125,11 @@ TEST(ObstacleFootprint, Prepare)
 
   auto node = nav2_util::LifecycleNode::make_shared("costmap_tester");
 
-  auto costmap_ros = std::make_shared<nav2_costmap_2d::Costmap2DROS>("test_global_costmap");
+  std::string ns = "/ns";
+  auto costmap_ros = std::make_shared<nav2_costmap_2d::Costmap2DROS>("test_global_costmap", ns);
   costmap_ros->configure();
 
   std::string name = "name";
-  std::string ns = "ns";
   critic->initialize(node, name, ns, costmap_ros);
```

Comment:
This test started failing, I'm not sure the fact that two nodes in different namespaces could communicate effectively was a feature or a bug, but now they need to have the same namespace. I added a convenience method for building a `Costmap2DROS` object while only specifying the parent_namespace to make this a bit less verbose, but I can revert that if needed